packages/ruleset/src/functions/response-status-codes.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * This function performs a few checks on each operation's responses field:
+ * 1. Status code 400 should be used instead of 422.
+ * 2. Status code 303 or 307 should be used instead of 302.
+ * 3. Operation responses should include at least one successful (2xx) status code.
+ * 4. Operation responses should not include status code 101 when successful (2xx) status codes are present.
+ * @param {*} operation an operation within the API definition
+ * @param {*} options unused, but needed as a placeholder due to the rule calling conventions
+ * @param {*} path the array of path segments indicating the "location" of the operation within the API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function responseStatusCodes(operation, options, { path }) {
+  if (!operation.responses) {
+    return [];
+  }
+
+  const [statusCodes, successCodes] = getResponseCodes(operation.responses);
+
+  if (statusCodes.length) {
+    const errors = [];
+
+    // Check for discouraged status codes per IBM API Handbook.
+    for (const statusCode of statusCodes) {
```

Comment:
You may be able to use `statusCodes.includes(...)` to avoid the loop here. Not sure if that would miss something and the loop is definitely good as is, but thought I'd mention it as an option

Reply:
yep, I was clinging to the old implementation too much :)
I've simplified the function quite a bit.